enclave/src/channel_storage.rs

```diff
@@ -31,17 +31,41 @@ lazy_static! {
         Arc::new(SgxMutex::new(None));
 }
 
+lazy_static! {
+    static ref GLOBAL_CHANNEL_STORAGE_TEST: Arc<SgxMutex<Option<ChannelStorage>>> =
+        Arc::new(SgxMutex::new(None));
+}
+
 #[derive(Clone)]
 pub struct ChannelStorage {
     pub sender: Sender<ChannelType>,
 }
 
 pub fn create_channel_get_receiver() -> Result<Receiver<ChannelType>, ChannelStorageError> {
+    internal_create_channel_get_receiver(false)
+}
+
+pub fn mock_create_channel_get_receiver() -> Result<Receiver<ChannelType>, ChannelStorageError> {
+    internal_create_channel_get_receiver(true)
+}
+
+fn internal_create_channel_get_receiver(
+    mock: bool,
```

Comment:
If this works it's fine for now I guess. But generally I think it's nicer to create a trait defining the function and implement the function two different structs. One mock struct and one real implementation.

Reply:
With a trait impl you could remove all these if statements within the functions.